app/src/main/java/org/usfirst/team25/scouting/calculatorapp/MainActivity.java

```diff
@@ -105,30 +105,35 @@ public void onClick(View view) {
             }
         });
 
-        for(final Button operationButton : operationButtons){
-            operationButton.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View view) {
-                    if(currentOperation != null)
-                        executeCalculation();
 
-                    // We use casework here and in the implementation of executeCalculation()
-                    // because the functions are essentially the same aside from the operation
-                    if(operationButton.getText().equals(getString(R.string.plus_label))){
-                        currentOperation = Operation.ADDITION;
-                    }
-                    else if(operationButton.getText().equals("x")){
-                        currentOperation = Operation.MULTIPLICATION;
-                    }
+            for (final Button operationButton : operationButtons) {
+                operationButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        if (currentOperation != null)
+                            if (inputNewNum == false) {
```

Comment:
Logic looks good here, but this could be combined with the `if` statement on the previous line so the code is more concise